examples/platforms/da15000/radio.c

```diff
@@ -31,528 +31,344 @@
-    {
-        s_extended_address[i] = address[i];
-    }
-
-    FTDF_setValue(FTDF_PIB_EXTENDED_ADDRESS,  s_extended_address);
+    FTDF_setValue(FTDF_PIB_EXTENDED_ADDRESS, address);
 }
 
 void otPlatRadioSetShortAddress(otInstance *aInstance, uint16_t address)
 {
     (void)aInstance;
 
-    s_short_address = address;
     FTDF_setValue(FTDF_PIB_SHORT_ADDRESS, &address);
 }
 
-
-void da15000RadioInit(void)
-{
-    /* Wake up power domains */
-    REG_CLR_BIT(CRG_TOP, PMU_CTRL_REG, FTDF_SLEEP);
-
-    while (REG_GETF(CRG_TOP, SYS_STAT_REG, FTDF_IS_UP) == 0x0);
-
-    REG_CLR_BIT(CRG_TOP, PMU_CTRL_REG, RADIO_SLEEP);
-
-    while (REG_GETF(CRG_TOP, SYS_STAT_REG, RAD_IS_UP) == 0x0);
-
-    REG_SETF(CRG_TOP, CLK_RADIO_REG, FTDF_MAC_ENABLE, 1);
-    REG_SETF(CRG_TOP, CLK_RADIO_REG, FTDF_MAC_DIV, 0);
-
-    phy_power_init();
-    hw_rf_system_init();
-    hw_rf_set_recommended_settings();
-    hw_rf_iff_calibration();
-
-    hw_rf_modulation_gain_calibration(0);  //RF MODE 0
-    hw_rf_dc_offset_calibration();
-
-    // ad_ftdf_init_phy_api();
-    FTDF_pib.CCAMode = 2;
-    ad_ftdf_init_phy_api_V2();
-    ad_ftdf_init_lmac();
-}
-
-
 ThreadError otPlatRadioEnable(otInstance *aInstance)
 {
-    First_Instace = aInstance;
-    (void)aInstance;
+    uint8_t DefaultChannel;
+    uint8_t EnableRX;
+    uint8_t MaxRetries;
+
     ThreadError error = kThreadError_None;
-    uint16_t DefaultChannel = DEFAULT_CHANNEL;
-    uint8_t value = 1;
+    VerifyOrExit(RadioState == kStateDisabled, error = kThreadError_InvalidState);
 
-    FTDF_setValue(FTDF_PIB_RX_ON_WHEN_IDLE, &value); //Wake();
-    FTDF_setValue(FTDF_PIB_CURRENT_CHANNEL, &DefaultChannel); //SetChannel(channel);
+    ThreadInstance = aInstance;
+    TransmitFrame.mPsdu = TransmitPsdu;
 
-    uint32_t max_retries = 0;
-    FTDF_setValue(FTDF_PIB_MAX_FRAME_RETRIES, &max_retries);
+    EnableRX = 1;
+    FTDF_setValue(FTDF_PIB_RX_ON_WHEN_IDLE, &EnableRX);
+    DefaultChannel = DEFAULT_CHANNEL;
+    FTDF_setValue(FTDF_PIB_CURRENT_CHANNEL, &DefaultChannel);
+    MaxRetries = 0;
+    FTDF_setValue(FTDF_PIB_MAX_FRAME_RETRIES, &MaxRetries);
 
     FTDF_Bitmap32 options =  FTDF_TRANSPARENT_ENABLE_FCS_GENERATION;
     options |= FTDF_TRANSPARENT_WAIT_FOR_ACK;
     options |= FTDF_TRANSPARENT_AUTO_ACK;
 
     FTDF_enableTransparentMode(FTDF_TRUE, options);
     otPlatRadioSetPromiscuous(aInstance, false);
+    RadioState = kStateSleep;
 
-    s_receive_frame.mPsdu = s_receive_message.mPsdu;
-    s_transmit_frame.mPsdu = s_transmit_message.mPsdu;
-    s_ack_frame.mPsdu = s_ack_message.mPsdu;
-    s_state = kStateReceive;
+exit:
     return error;
 }
 
 ThreadError otPlatRadioDisable(otInstance *aInstance)
 {
     (void)aInstance;
-    s_state = kStateDisabled;
+
+    RadioState = kStateDisabled;
+
     return kThreadError_None;
 }
 
 bool otPlatRadioIsEnabled(otInstance *aInstance)
 {
     (void)aInstance;
-    return (s_state != kStateDisabled);
+
+    return (RadioState != kStateDisabled) ? true : false;
```

Comment:
You can simply return ``(RadioState != kStateDisabled)`` without the inline if statement.